### DISCUSSION TITLE
UI Consideration: macOS - remove media window rounded corners

### DISCUSSION THREAD

**Discussion by riggles** [sircharlo/meeting-media-manager#425] [Ideas] [answered]
In macOS, all application windows have rounded corners. The same is true for media in M3 when displayed in a window. When screen sharing that media window on Zoom, those transparent corners are captured as solid white filled areas. It stands out against the black year text background or letterboxed content.

The media playback window of VLC on macOS has square corners, so it displays clean for participants. Is it possible to incorporate the same UI behavior in M3 so that full-screen and windowed media playback on both macOS and Windows all have the same presentation during screen shares?

Thank you for considering, and thank you for all the effort you've already put into this tool that has really streamlined the logistics of media playback at meetings.

**Comment by sircharlo** [comment]
You're correct! I've pushed a commit that should remove these rounded corners on macOS [here](https://github.com/sircharlo/meeting-media-manager/commit/4c5251c2c957287825bc790f17b466e177fc8a82#diff-58417e0f781b6656949d37258c8b9052ed266e2eb7a5163cad7b0863e6b2916aR209). On Windows they were already showing up as square, so this went under my radar. Will be released in the next update.

**Comment by mtdvlpr** [comment]
This is unfortunately not possible to fix due to [a limitation in Electron](https://github.com/electron/electron/issues/36251)

**Comment by mtdvlpr** [comment]
Hi @riggles, I've been thinking about a possible fix for this, but just to be sure: the rounded corners are only present when the media window is a window and not when it's in fullscreen, correct?

**Reply by riggles** [reply]
Correct, only in windowed mode.

**Reply by mtdvlpr** [reply]
@riggles, alright. In that case, I have a potential fix for you to test. You can find it [here](https://drive.google.com/drive/folders/1_2CzKieYGaK3LcYrEwzNQgfTpNReOSx9?usp=sharing). Let me know if this (finally) fixes things!

**Reply by mtdvlpr** [reply]
Please try all kinds of actions (since I can't test on MacOS myself), like toggling the preferred output, toggling the media presentation mode, hiding/showing the media window, restarting M3 with both fullscreen and windowed mode.

**Reply by riggles** [reply]
Thank you for trying to make this adjustment. The version you pushed does show square corners in windowed mode. Excellent. I noticed the following behaviors:

- If using Moveable Window, entering Settings from the Home Screen will reset the presentation mode to fullscreen (Screen 2, in my case).
- Disabling the presentation mode preference has two different behaviors, depending on the mode you're in: 1) If in windowed mode, the year text window will remain up on the main display. Then when turning the preference back on, the mode selection will default to Screen 2, but the year text will remain in a window on the main display. I need to set the mode to Movable Window, and then back to Screen 2 for fullscreen to be triggered. 2) If disabling the preference while in fullscreen mode, the year text will go away completely - nothing on Screen 2 and no window. Turning the preference back on will display the year text on Screen 2 as expected. 
- Hiding/showing the media window works as expected.
- Restarting M3 will retain the selected presentation mode properly for both.

Hope the testing results help.

**Reply by mtdvlpr** [reply]
Hi @riggles, thank you! The feedback is very helpful. I think I fixed your second point of feedback. I updated the file in the link I sent previously. Your first point already existed and is a side effect of having a second screen as the default. I have yet to find a solution for that.

**Reply by riggles** [reply]
@mtdvlpr The update fixed that second issue I mentioned. Thank you very much for your help.

**Comment by mtdvlpr** [comment, marked as answer]
Fixed (again) in #999 and will be in the next release :)
